aspnetcore/security/authorization/policies.md

```diff
@@ -214,18 +214,37 @@ For example, the previous `BadgeEntryHandler` could be rewritten as follows:
 
 The `HandleRequirementAsync` method you implement in an authorization handler has two parameters: an `AuthorizationHandlerContext` and the `TRequirement` you are handling. Frameworks such as MVC or SignalR are free to add any object to the `Resource` property on the `AuthorizationHandlerContext` to pass extra information.
 
+::: moniker range=">= aspnetcore-5.0"
+
+When using endpoint routing, authorization is typically handled by the Authorization Middleware. In this case, the `Resource` property is an instance of <xref:Microsoft.AspNetCore.Http.HttpContext>. The context can be used to access the current endpoint, which can be used to probe the underlying resource to which you're routing. For example:
+
+```csharp
+if (context.Resource is HttpContext httpContext)
+{
+    var endpoint = httpContext.GetEndpoint();
+    var actionDescriptor = endpoint.Metadata.GetMetadata<ControllerActionDescriptor>();
+    ...
+}
+```
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-5.0"
```

Comment:
Because this moniker is already inside a >= 3.0 moniker, I'm assuming this will be treated as a combination of >= 3.0 and < 5.0. 

Let me know if that's incorrect.

Reply:
It breaks :boom:. They can't be merged in any way. You have to flip in and out of each range, which can get very messy and hard to maintain quickly. That's why we usually end up going with *whole-topic versioning* (complete copy with changes) ... in spite of the terrible duplication of coverage that it entails. It does have one nice benefit tho ... when dropping a version ... it's just a delete of the whole versioned block.

Reply:
OK, thanks for the explanation @guardrex. I'll update this PR later to use whole-topic versioning. Does that still cause problems with cross-topic heading links, where there's a - 1 suffix on the duplicated anchors, for example?

Reply:
Yes, it does. Versioning was never designed for how we're using it, and I haven't had time to follow up on Scott's guidance about activating *version-by-file*, which **_IS_** designed (AFAIK) to solve that problem. I have a handful of Blazor topics itch'in to use *version by file* if a test of it is successful. I hope to get to that soon-ish.

Personally, I've solved that cross-link problem in the simplest way ... I just link to the topic. IMO, these topics generally aren't long enough to sweat breaking cross-links and fighting with a sour build report. By linking the topic, devs need to find the section, so I usually go with something like ...

> See the \*My Cool Heading\* section of the \<xref:blah/blah/blah> article.

No bookmark ... no :boom: ... and they just need to select the named section from the sidebar.

This is NOT policy tho. It's just a nice quick fix in a half-dozen spots in Blazor topics to tide me over to *version-by-file* as a total solution. @Rick-Anderson will let you know how he wants it handled.

Also, duplicating sections duplicates `alt` text. I recommend taking the time to make `alt` text just a little different in each versioned block. Fortunately, we avoid images in non-tutorials, and that's turned out to be the right call IMHO.

Reply:
... and one more side-note on breaking linking ... it may throw in the build report but work in the live topic where a section is cross-linked in the doc. Rick likes the anchor approach, but that's not HTML5 friendly. I prefer (again, NOT policy here) to place the `id` in heading links `<hX>` and link within version sections to the right named heading section.

Version 5.x section ...

```
See the [Blah Blah Blah](#blah-blah-blah-5x) section of this article.

<h2 id="blah-blah-blah-5x">Blah Blah Blah</h2>
```

Version 3.x section ...

```
See the [Blah Blah Blah](#blah-blah-blah-3x) section of this article.

<h2 id="blah-blah-blah-3x">Blah Blah Blah</h2>
```

The build system ignores HTML. Those are HTML5 friendly. Those don't break. There is still the problem of cross-doc links; but like I said, I'll just say "See the *Blah Blah Blah* section of the \<xref:...> article." to solve that.

Reply:
There might only be **one** spot where I did that. See it in use here :point_right: https://github.com/dotnet/AspNetCore.Docs/edit/main/aspnetcore/blazor/blazor-server-ef-core.md

Reply:
That all makes sense, thanks. I'll switch this to whole-topic versioning and then have a look at if/where sections from this topic are referenced and report back.

Reply:
We should figure out version by file sometime, but whole-topic works pretty well.

Reply:
I see a few heading references from outside of the topic:

- fundamentals\routing.md:
   - `* [Apply policies using the [Authorize] attribute](xref:security/authorization/policies#apply-policies-to-mvc-controllers)`
- migration\22-to-30.md
   - `If the app uses custom [authorization handlers](xref:security/authorization/policies#authorization-handlers), endpoint routing passes a different resource type to handlers than MVC. Handlers that expect the authorization handler context resource to be of type <xref:Microsoft.AspNetCore.Mvc.Filters.AuthorizationFilterContext> (the resource type [provided by MVC filters](xref:security/authorization/policies#access-mvc-request-context-in-handlers)) will need to be updated to handle resources of type <xref:Microsoft.AspNetCore.Routing.RouteEndpoint> (the resource type given to authorization handlers by endpoint routing).`

- security\authorization\claims.md
   - `If the claim value isn't a single value or a transformation is required, use [RequireAssertion](/dotnet/api/microsoft.aspnetcore.authorization.authorizationpolicybuilder.requireassertion). For more information, see [Use a func to fulfill a policy](xref:security/authorization/policies#use-a-func-to-fulfill-a-policy).`

- security\authorization\dependencyinjection.md
   - `[Authorization handlers must be registered](xref:security/authorization/policies#handler-registration) in the service collection during configuration using [dependency injection](xref:fundamentals/dependency-injection).`

- security\authorization\resourcebased.md
   - `Writing a handler for resource-based authorization isn't much different than [writing a plain requirements handler](xref:security/authorization/policies#security-authorization-policies-based-authorization-handler). Create a custom requirement class, and implement a requirement handler class. For more information on creating a requirement class, see [Requirements](xref:security/authorization/policies#requirements).`

Which approach should I take for these? The topic already has a few `<a name="...">` tags, so shall I go with that for consistency?

Reply:
> Which approach should I take for these? The topic already has a few `<a name="...">` tags, so shall I go with that for consistency?

I prefer  `<a name="...">` tags as they're obviously immutable. Just fixed a bug yesterday where an H2 changed and broke a link.

Reply:
:cry: lol